lib/Dialect/FIRRTL/InnerSymbolTable.cpp

```diff
@@ -186,17 +223,19 @@ Operation *InnerRefNamespace::lookupOp(hw::InnerRefAttr inner) {
 }
 
 //===----------------------------------------------------------------------===//
-// InnerRef verification
+// InnerRefNamespace verification
 //===----------------------------------------------------------------------===//
 
 namespace detail {
 
-LogicalResult verifyInnerRefs(Operation *op) {
+LogicalResult verifyInnerRefNamespace(Operation *op) {
   // Construct the symbol tables.
   InnerSymbolTableCollection innerSymTables;
   SymbolTable symbolTable(op);
   InnerRefNamespace ns{symbolTable, innerSymTables};
-  innerSymTables.populateTables(op);
+
+  if (failed(innerSymTables.populateAndVerifyTables(op)))
+    return failure();
```

Comment:
Makes more sense to me to populate the table before constructing the `InnerRefNamespace` from it.

Reply:
Good call, thanks!